application/backend/app/services/event/event_bus.py

```diff
@@ -1,42 +1,28 @@
 # Copyright (C) 2025 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
+from collections.abc import Callable, Sequence
+from enum import StrEnum
 from multiprocessing.synchronize import Condition
+from threading import RLock
 
 
-class Listener:
-    """
-    Events listener, gets callback notifications upon certain events, i.e. source/sink change, pipeline update.
-    """
-
-    def on_source_changed(self) -> None:
-        """
-        Callback method for source changes.
-        """
-
-    def on_sink_changed(self) -> None:
-        """
-        Callback method for sink changes.
-        """
-
-    def on_pipeline_dataset_collection_policies_changed(self) -> None:
-        """
-        Callback method for dataset collection policies changes.
-        """
-
-    def on_pipeline_status_changed(self) -> None:
-        """
-        Callback method for pipeline status changes.
-        """
+class EventType(StrEnum):
+    SOURCE_CHANGED = "SOURCE_CHANGED"
+    SINK_CHANGED = "SINK_CHANGED"
+    PIPELINE_DATASET_COLLECTION_POLICIES_CHANGED = "PIPELINE_DATASET_COLLECTION_POLICIES_CHANGED"
+    PIPELINE_STATUS_CHANGED = "PIPELINE_STATUS_CHANGED"
 
 
 class EventBus:
     def __init__(
         self, source_changed_condition: Condition | None = None, sink_changed_condition: Condition | None = None
     ) -> None:
-        self._listeners: list[Listener] = []
+        self._event_handlers: dict[EventType, list[Callable]] = {}
```

Comment:
If we init it as `defaultdict(list)`, there won't be need to use `.setdefault(event_type, [])` in `subscribe`